src/tests/ftest/cart/test_group.c

```diff
@@ -204,6 +204,25 @@ client_cb_common(const struct crt_cb_info *cb_info)
 		sem_post(&test_g.t_token_to_proceed);
 		D_ASSERT(rpc_req_output->bool_val == true);
 		break;
+	case TEST_OPC_SWIM_STATUS:
+		rpc_req_input = crt_req_get(rpc_req);
+		if (rpc_req_input == NULL)
+			return;
```

Comment:
change to asserts, this is fatal if input is null

Reply:
It looks like all test_group.c was moved to [test_group_np_common.c](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/cart/test_group_np_common.h), but the [test_group.c](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/cart/test_group.c) file was left around.  It's not used in [the SConscript file](https://github.com/daos-stack/daos/blob/4f888609569c0a9ff95b4e249a651056081459e7/src/tests/ftest/cart/SConscript#L53).  I'm looking to apply your recommendations to test_group_np_common.c, and remove test_group.c to avoid confusion.

Reply:
581ebf161f1f6ab82a4ba2b619dbf33a31e5b92d